spec/models/filters/criteria/filter_for_disabilities_spec.rb

```diff
@@ -102,4 +115,43 @@
       end
     end
   end
+
+  describe 'HIV/AIDS visibility' do
+    context 'when user has permission to view HIV/AIDS status' do
+      before do
+        allow(user).to receive(:can_view_hiv_status?).and_return(true)
+      end
+
+      it 'includes HIV/AIDS in available disabilities' do
+        expect(filter.available_disabilities).to include('HIV/AIDS')
+      end
+
+      it 'applies HIV/AIDS filter correctly' do
+        filter.disabilities = [8] # HIV/AIDS
+        result = criteria.apply(scope)
+        expect(result.pluck(:id)).to contain_exactly(enrollments[4].id)
+      end
+    end
+
+    context 'when user does not have permission to view HIV/AIDS status' do
+      before do
+        allow(user).to receive(:can_view_hiv_status?).and_return(false)
+      end
+
+      it 'excludes HIV/AIDS from available disabilities' do
+        expect(filter.available_disabilities).not_to include('HIV/AIDS')
+      end
+
+      it 'removes HIV/AIDS from selected disabilities' do
+        filter.disabilities = [8] # HIV/AIDS
```

Comment:
it looks like attribute assignment doesn't hit the filter. The filter only occurs if you use the update method:
```suggestion
        filter.update(disabilities: disabilities + [8]) # HIV/A
```

Reply:
Thanks, I swear I reviewed this earlier, but I guess process wasn't very good.